### ISSUE TITLE
samtoh5 "Segmentation fault (core dumped)"

### ISSUE THREAD

**Issue by gawbul** [PacificBiosciences/blasr#62]
I get the following error when trying to convert a sam to an h5 file.

```
$ samtoh5 Output/pbreads_to_pbasm_pbalign.sam Seq_Data/PacBio/PB_assembly/PB_polished.fa Output/pbreads_to_pbasm_pbalign.cmp.h5
[INFO] 2015-06-24T11:49:11 [samtoh5] started.
Segmentation fault (core dumped)
```

**Comment by ylipacbio** [comment]
Would you please let me know git commit number, OS, gcc, hdf5 version? 

Please also send the *.sam and *.fa files, or a small example files which broke.

Thanks

**Comment by jayhesselberth** [comment]
I'm getting the same core dump

I'm on:

- gcc commit `87891a5d9d963a6f6e206f845964d3dc2f50eeba`
- Linux amc-tesla 2.6.32-131.0.15.el6.x86_64 #1 SMP Tue May 10 15:42:40 EDT 2011 x86_64 x86_64 x86_64 GNU/Linux
- $ gcc --version
gcc (GCC) 5.1.0
- hdf5 1.8.15-patch1

Not sure what you want for sam file ... but can post fa

**Comment by ylipacbio** [comment]
I want the first two args for samtoh5

e.g., in gawbul's case, 
```Output/pbreads_to_pbasm_pbalign.sam``` and ```Seq_Data/PacBio/PB_assembly/PB_polished.fa```

Thanks

**Comment by jayhesselberth** [comment]
I'll have to regenerate these files, keeping the temp files from the run with the current `samtoh5`.

In the meantime, I'm using an older `samtoh5`, which is working fine in the updated `pbalign` pipeline.

And I compiled `hdf5 1.8.15-patch1` from source with `gcc 5.1.0`.

**Comment by gawbul** [comment]
Apologies for the delay. I'm on:
```
[smoss@biolserva blasr]$ git rev-parse HEAD
463a3697c170c3ba15421b32d51e5ca82e8ece8d

[smoss@biolserva ~]$ uname -a
Linux biolserva 3.19.8-100.fc20.x86_64 #1 SMP Tue May 12 17:08:50 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux

[sm1818@biolserva ~]$ cat /etc/redhat-release 
Fedora release 20 (Heisenbug)

[smoss@biolserva ~]$ gcc --version
gcc (Homebrew gcc 5.1.0) 5.1.0
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.

[smoss@biolserva blasr]$ strings ~/.linuxbrew/Cellar/hdf5/1.8.14/lib/libhdf5.a | grep "HDF5 library version:"
HDF5 library version: 1.8.14
HDF5 library version: %d.%d.%d
HDF5 library version: 1.8.14
```

**Comment by gawbul** [comment]
Have recompiled hdf5 (1.8.14) and blasr (2.2) with gcc 4.8.4 and this seems to have fixed the seg fault?

**Comment by pb-cdunn** [comment]
Closing since gawbul solved this by re-installing hdf5 consistently.

**Comment by shashidhar22** [comment]
Having the same issue with samtoH5, recompiled blasr and hdf5 with gcc 6, blasr works perfectly fine, samtoh5 gives a segmentation fault error. Is there no other fix but to compile blasr and hdf5 with gcc 4.8?
